### PR TITLE
Introduce rebar_lock_deps_plugin to pin dep versions

### PR DESCRIPTION
**Pull request by kuenishi**
...to make packages.

### CODE REVIEW

**Comment by kuenishi** [comment]
After this merged, I'll tag 0.10.1 make some packages for our upcoming CS testing.

File: rebar.config [hunk 1]
```diff
@@ -29,12 +29,16 @@
...
    {git, "git://github.com/nifoc/katja.git",
-   {branch, "master"}}}
+   {branch, "master"}}},
+  {rebar_lock_deps_plugin, ".*", {git, "git://github.com/seth/rebar_lock_deps_plugin.git", {tag, "3.1.0"}}}
  ]}.
 
 {erl_opts, [{src_dirs, [src]},
            {parse_transform, lager_transform}]}.
 
+{plugin_dir, ".plugins"}.
```
Comment (shino) [inline rebar.config]: There is no directory named `.plugins`. in git. Is this config needed? or generated?

File: Makefile [hunk 2]
```diff
@@ -21,6 +21,19 @@ rel: deps compile
 deps:
 	./rebar get-deps
 
+##
+## Lock Targets
```
Comment (shino) [inline Makefile]: These targets should be included in `.PHONY`

**Comment by shino** [comment]
Not all sample configs are copied. I think these samples are helpful even in server environment at starting point.
I guess rebar's reltool overlay function has copy files in a directory, e.g.

```
{overlay, "../examples/", "etc/}
```

**Comment by shino** [comment]
In making packages, `deps` target is used:
https://github.com/basho/basho_bench/blob/master/Makefile#L72

I don't test it actually, but is it intended behavior?

**Comment by slfritchie** [comment]
@kuenishi Is the `rebar.config.lock` file supposed to be added to this PR?

**Comment by kuenishi** [comment]
@slfritchie Yes, it should be added. There's a lot to fix here.

**Comment by kuenishi** [comment]
Updated. Preset configs should be with another pull request.

**Comment by shino** [comment]
Confirmed:

- `make package` uses locked dependencies and
- package is successfully created on Ubuntu 14.04

:+1:

**Pull request merged**